client.go

```diff
@@ -724,13 +714,17 @@ func (c *TunaSessionClient) newSession(remoteAddr string, sessionID []byte, conn
 		if err != nil {
 			log.Println("Write message error:", err)
 			conn.Close()
-			return ncp.ErrConnClosed
+			return err // ncp.ErrConnClosed
```

Comment:
ncp will handle ErrConnClosed differently

Reply:
If return ncp.ErrConnClosed to NCP, NCP connection would be closed, and reconnecting would not happen even Tuna session connection reconnected.